stacks-signer/src/chainstate.rs

```diff
@@ -453,32 +430,94 @@ impl SortitionsView {
         Ok(true)
     }
 
-    fn check_tenure_change_block_confirmation(
+    /// Check if the tenure change block confirms the expected parent block (i.e., the last globally accepted block in the parent tenure)
+    fn check_tenure_change_confirms_parent(
         tenure_change: &TenureChangePayload,
         block: &NakamotoBlock,
         signer_db: &SignerDb,
     ) -> Result<bool, ClientError> {
-        // in tenure changes, we need to check:
-        // (1) if the tenure change confirms the expected parent block (i.e.,
-        //     the last block we signed in the parent tenure)
-        // (2) if the parent tenure was a valid choice
-        Self::confirms_known_blocks_in(block, &tenure_change.prev_tenure_consensus_hash, signer_db)
+        let Some(last_globally_accepted_block) = signer_db
```

Comment:
This would only work reliably if we're guaranteed that the signer has seen enough signatures for the block, since that appears to be the only way the signer DB would mark a block as globally accepted.  But, I don't think the code makes this guarantee.  It's possible that signature acceptance messages don't arrive on the node in time.  Specifically, this can happen:

Suppose signers S[0]...S[9] each have an equal share of signing power, and each signer S[i] is connected to node N[i].  Let M be the miner. Then, suppose the following happens:

1. Signers S[7]..S[9] and nodes N[7]...N[9] go to sleep / get partitioned
1. M produces a block proposal P
1. Signers S[0]...S[6] all produce signatures for P
1. M receives signatures from S[0]...S[6] and broadcasts the block.  S[0]...S[6] also see the signatures, and broadcast the block B[0].
1. A new Bitcoin block arrives, and miner M wins sortition
1. Miner M produces a tenure-change block proposal and signers S[0]...S[6] produce signatures.  M and S[0]...S[6] see all of the signatures and broadcast the tenure-change block B[1]
1. N[0]...N[6] each process block B[1], which causes them to clear out their miner and signer StackerDBs (since they get reconfigured on each tenure boundary)
1. N[7]...N[9] and S[7]...S[9] wake up / rejoin the network
1. N[7]...N[9] process the next Bitcoin block, Stacks block B[0], and tenure-change Stacks block B[1]. 
1. N[7]...N[9] reconfigure their StackerDBs when the process B[1]

Do signers S[7]...S[9] see the messages from M and S[0]...S[6] for B[0]? Because, I don't think they do, meaning that S[7]...S[9] won't ever mark B[0] as globally accepted.

Reply:
I think instead, one of two things needs to happen:

* the signer needs to query the node for the last globally accepted block in the given tenure, or
* the signer needs to capture Nakamoto block events from the node, and mark Nakamoto blocks as globally accepted

I'd prefer the first option, because the second option, while saving an RPC, introduces the possibility that the signer's DB gets out of sync with the chainstate (e.g. suppose the node operator shut down the signer and removed its event observer entry for a time, and then re-added it).

Reply:
Yeah the RPC endpoint is my preferred option too.

Reply:
One other wrinkle here that the API endpoint can address -- the highest globally accepted block for the first-ever Nakamoto block is going to be an epoch2 block.